code/game/machinery/modular_shield.dm

```diff
@@ -0,0 +1,681 @@
+	var/obj/machinery/modular_shield/module/node/connected_node
+
+	///This is the turf that we are facing and able to search for connections through
+	var/turf/connected_turf
+
+/obj/machinery/modular_shield/module/Initialize(mapload)
+	. = ..()
+
+	connected_turf = get_step(loc, dir)
+
+/obj/machinery/modular_shield/module/Destroy()
+
+	if(shield_generator)
+		LAZYREMOVE(shield_generator.connected_modules, (src))
+		shield_generator.calculate_boost()
+	if(connected_node)
+		LAZYREMOVE(connected_node.connected_through_us, (src))
+	return ..()
+
+/obj/machinery/modular_shield/module/examine(mob/user)
+	. = ..()
+
+	if(isnull(shield_generator) && isnull(connected_node))
+		. += "It can be loosened and rotated with a screwdriver and wrench. It can be connected to a node or generator with a multitool."
+		return
+	. += "It can be loosed and rotated with a screwdriver and wrench, rotating it will sever its connection."
+
+/obj/machinery/modular_shield/module/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+
+	panel_open = !(panel_open)
+	tool.play_tool_sound(src, 50)
+	update_icon_state()
+	if(panel_open)
+		to_chat(user, span_notice("You open the maintenance hatch of [src]."))
+		return TRUE
+	to_chat(user, span_notice("You close the maintenance hatch of [src]."))
+	return TRUE
+
+/obj/machinery/modular_shield/module/multitool_act(mob/living/user, obj/item/tool)
+	. = ..()
+
+	//rather than automatically checking for connections its probably alot less
+	//expensive to just make the players manually multi tool sync each part
+	try_connect(user)
+	return TRUE
+
+/obj/machinery/modular_shield/module/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+
+	if(default_change_direction_wrench(user, tool))
+		if(shield_generator)
+			LAZYREMOVE(shield_generator.connected_modules, (src))
+			shield_generator.calculate_boost()
+			shield_generator = null
+			update_icon_state()
+		if(connected_node)
+			LAZYREMOVE(connected_node.connected_through_us, (src))
+			connected_node = null
+		connected_turf = get_step(loc, dir)
+		return TRUE
+
+/obj/machinery/modular_shield/module/crowbar_act(mob/living/user, obj/item/tool)
+	. = ..()
+
+	if(default_deconstruction_crowbar(tool))
+		return TRUE
+
+
+/obj/machinery/modular_shield/module/setDir(new_dir)
+	. = ..()
+	connected_turf = get_step(loc, dir)
+
+///checks for a valid machine in front of us and connects to it
+/obj/machinery/modular_shield/module/proc/try_connect(user)
+
+	if(shield_generator || connected_node)
+		balloon_alert(user, "already connected to something!")
+		return
+
+	shield_generator = (locate(/obj/machinery/modular_shield_generator) in connected_turf)
+
+	if(shield_generator)
+
+		LAZYOR(shield_generator.connected_modules, (src))
+		balloon_alert(user, "connected to generator")
+		update_icon_state()
+		if(istype(src, /obj/machinery/modular_shield/module/node))
+			var/obj/machinery/modular_shield/module/node/connected_node = src
+			connected_node.connect_connected_through_us()
```

Comment:
We should never be checking src for type, you should move this to its subtype instead.

Reply:
done